hrcase/CRM/HRCase/Upgrader.php

```diff
@@ -248,16 +269,144 @@ public static function isExtensionEnabled($key)  {
   }
 
   /**
+   * Removes a list of defined activity types for a given component
+   *
+   * @param array $activityTypes
+   *   A list of activity types names to remove
+   * @param int $componentName
+   *   (e.g : CiviCase, CiviTask .. etc)
+   */
+  private function removeActivityTypesList($activityTypes, $componentName) {
+    civicrm_api3('OptionValue', 'get', [
+      'name' => ['IN' => $activityTypes],
+      'component_id' => $componentName,
+      'option_group_id' => "activity_type",
+      'api.OptionValue.delete' => ['id' => '$value.id'],
+    ]);
+  }
+
+  /**
+   * Removes a list of unused ( or unneeded ) case and activity types
+   * managed records
+   *
+   * @param array $caseTypes
+   * @param array $activityTypes
+   */
+  private function up1402_removedUnusedManagedEntities($caseTypes, $activityTypes) {
+    $entitiesToRemove['civicase:act:Background Check'] = 'OptionValue';
+
+    foreach (array_merge($caseTypes, ['Appraisal', 'Probation']) as $caseType) {
+      $entitiesToRemove[$caseType] = 'CaseType';
+    }
+
+    foreach ($activityTypes as $activityType) {
+      $entitiesToRemove['civitask:act:' . $activityType] = 'OptionValue';
+    }
+
+    foreach ($entitiesToRemove as $name => $entity) {
+      $extension = 'org.civicrm.hrcase';
+      if ($name == 'civicase:act:Background Check') {
+        $extension = 'civicrm';
+      }
+
+      $this->removeManagedEntityRecord($extension, $name, $entity);
+    }
+  }
+
+  /**
+   * Removes unused ( or unneeded ) case types
+   */
+  private function up1402_removeUnusedCaseTypes() {
+    // Remove (Appraisal) & (Probation) case types
+    $caseTypesToRemove = ['Appraisal', 'Probation'];
+
+    foreach($caseTypesToRemove as $caseType) {
+      $this->removeUnusedCaseType($caseType);
+    }
+  }
+
+  /**
+   * Creates/Updates default case types to be shipped
+   * with this extension
+   *
+   * @param $defaultCaseTypes
+   */
+  private function up1402_createDefaultCaseTypes($defaultCaseTypes) {
+    foreach ($defaultCaseTypes as $caseType) {
+      $type = civicrm_api3('CaseType', 'get', [
+        'sequential' => 1,
+        'name' => $caseType['name'],
+        'limit' => 1
+      ]);
+
+      if (!empty($type['id'])) {
+        $caseType['id'] = $type['id'];
+      }
+
+      civicrm_api3('CaseType', 'create', $caseType);
+    }
+  }
+
+  /**
+   * Creates default activity types to be shipped
+   * with this extension if they are not exits
+   *
+   * @param $defaultActivityTypes
+   */
+  private function up1402_createDefaultActivityTypes($defaultActivityTypes) {
+    foreach ($defaultActivityTypes as $activityType) {
+      $type = civicrm_api3('OptionValue', 'get', [
+        'sequential' => 1,
+        'name' => $activityType,
+        'option_group_id' => "activity_type",
```

Comment:
please make sure to use single quotes for all the keys and values on this array and on the next one